core/templates/pages/admin-page/features-tab/admin-features-tab.component.ts

```diff
@@ -214,12 +212,14 @@ export class AdminFeaturesTabComponent implements OnInit {
       await this.apiService.updateFeatureFlag(
         feature.name, commitMessage, feature.rules);
 
-      this.setStatusMessage('Saved successfully.');
+      this.featureFlagNameToBackupMap.set(feature.name, cloneDeep(feature));
```

Comment:
Suggest also verifying that if the update fails, the in-memory backup is not changed.